### DISCUSSION TITLE
How RTP packets are handled in case of RTSP to RTSP proxy

### DISCUSSION THREAD

**Discussion by gizak** [bluenviron/mediamtx#3391] [Questions]
### Question

Hi I recently checked this awesome project. It is everything I need for my work (multiple protocols and stream routing, etc.). But currently I encountered a problem: I need to setup a RTSP server, with one end pushing the stream to it and read out from another end. The RTSP worked fine but it seems that all RTP packets were "reinterpreted" or translated and re-packed. 

Here is how packets sent (H264 mode0, no STAP, no FU)
<img width="1351" alt="Screenshot 2024-05-25 at 11 27 54" src="https://github.com/bluenviron/mediamtx/assets/2100512/2b55e7dc-6e82-4e97-a9d7-beb843945157">

Here is how packets read from mediamtx server:
 
<img width="1300" alt="Screenshot 2024-05-25 at 11 29 08" src="https://github.com/bluenviron/mediamtx/assets/2100512/9e19dca6-48a4-44c9-9b5d-da2a7f6446eb">

It seems that all sending packets were decoded and re-packed. But in my case I need H264 mode0 (packetization-mode=0, no aggregation, no STAP, no FU) to be compatible for some old receiver. How the RTP packets are handled from RTSP to RTSP? Is there a workaround to allow RTP passthru or similar?

**Comment by aler9** [comment]
Hello, the situation you are describing is caused by two separate issues: first, we are currently remuxing all packets that have a length greater than the UDP MTU. This allows us to produce streams that are compatible with almost any client. The second issue is that i looked around and i found out that we are currently not supporting H264 `packetization-mode=0`, therefore the remux is always in `packetization-mode=1`, but this is wrong.

I think the first step into solving this situation is to support `packetization-mode=0`. I've created a feature request here: https://github.com/bluenviron/gortsplib/issues/585

Then we'll try to see what to do with the automatic remuxing feature. Since FU-A packets are not supported with `packetization-mode=0`, we could disable that, but i need to think about this.

[a comment by an automated account is not shown]
